spec/game_spec.rb

```diff
@@ -0,0 +1,32 @@
+require "spec_helper"
+require "game"
+
+RSpec.describe Game do
+  class UserWithOneValidInput
+    def valid_input(message:, parse_input:, validator:)
+      1
+    end
+  end
+
+  class PlayerWithoutMethods end
+  class ComputerPlayerWithoutMethods end
+  class ParseInputWithNoMethods end
+  class ValidatorWithNoMethods end
+
+  describe "#players" do
+    it "returns an array of players based on the user's selection" do
+      computer_player = ComputerPlayerWithoutMethods.new
+      human_player = PlayerWithoutMethods.new
+      game_options = {
+        "You go first" => [human_player, computer_player],
+        "Computer goes first" => [computer_player, human_player],
+      }
+      parse_input = ParseInputWithNoMethods.new
+      validator = ValidatorWithNoMethods.new
+      user = UserWithOneValidInput.new
```

Comment:
Game has a lot of dependencies. If we had a new constraint that an object could only have at most 4 collaborators, how could we deal with that?

Reply:
I think your suggestion below makes sense of potentially passing parse_input into the user.